### ISSUE TITLE
Clarify how to fix ANN101 and ANN102

### ISSUE THREAD

**Issue by askoretskiy** [python-discord/flake8-annotations#82]
**Description**

I have read several times documentation and PEP 484 and PEP 563 but did not get a clear idea how to get rid of these.

It would be very helpful to see any example of fixing this warning.

Please provide a clear & concise description of the proposed feature.

**Rationale/Use Case**

I've defined a model in Django:

```python
from django.db import models

class Content(models.Model):
    name = models.CharField(max_length=255)

   def __str__(self) -> str:
       return self.name
```

That gives me a warning ANN101 that `self` is not annotated. That's strange, while it method of `Content` class... Okay, let's try this:

```python
from django.db import models
from typing import TypeVar

TContent = TypeVar("TContent", bound="Content")

class Content(models.Model):
    name = models.CharField(max_length=255)

   def __str__(self: TContent) -> str:
       return self.name
```

Does it suppose to look like this?  (BTW, what is the point of introducing `TContent` here?)

With this code PyCharm does not recognize `self.name` as a field defined in a model...

Now I'm wondering whether it is PyCharm issue or something wrong with linter.

At the same time, this code:

```
from __future__ import annotations
from django.db import models

class Content(models.Model):
    name = models.CharField(max_length=255)

   def __str__(self: Content) -> str:
       return self.name
```

It works and PyCharm recognizes `self.name` attribute. But it seems redundant to define it for each method...

**Solution**

Provide an example how to *correctly* fix the warnings ANN101 and ANN102. Thank you.

**Comment by sco1** [comment]
Hello, please see the discussion in #75

**Comment by askoretskiy** [comment]
I think the frustration comes from this part not clear from documentation. Maybe documentation should be improved (but I have no idea how :-)

**Comment by sco1** [comment]
There are multiple methods for accomplishing this and you've provided two valid ones, via PEP 484 and PEP 563, in your original post.

The problems seem to be:
  1. PyCharm doesn't resolve a bound `TypeVar`
  2. The annotations seem redundant

We have no control or influence over the former, and the latter is a decision for you to make.

As discussed in #75:

> The purpose of this package is not to enforce the related PEPs, as they make no mandates regarding full annotation coverage. Its purpose is to detect the absence of these annotations; all errors are exposed and we leave it up to the developer to determine what subset of these errors to enforce for their project. This project, for example, currently ignores 6 of its own errors.

Which also references a statement from PEP 484:

> In most cases the first argument of class and instance methods does not need to be annotated, and it is assumed to have the type of the containing class for instance methods, and a type object type corresponding to the containing class object for class methods.
